support/metrics_plaid.py

```diff
@@ -189,329 +35,79 @@ def plaid_kyc(acc, txn):
-
-            feedback['credit']['credit_limit'] = credit_lim
-        else:
-            raise Exception('no credit limit')
-
-    except Exception as e:
-        score = 0
-        feedback['credit']['error'] = str(e)
-
-    finally:
-        return score, feedback
-
-
-# @evaluate_function
-def credit_util_ratio(acc, txn, feedback, params):
-    '''
-    Description:
-        A score reflective of the user's credit utilization ratio, that is credit_used/credit_limit
-
-    Parameters:
-        acc (list): Plaid 'Accounts' product
-        txn (list): Plaid 'Transactions' product
-        feedback (dict): score feedback
-        params (dict): model parameters, i.e. coefficients
-
-    Returns:
-        score (float): score for avg percent of credit limit used
-        feedback (dict): score feedback
-    '''
-    try:
-        # Dynamically select best credit account
-        dynamic = dynamic_select(acc, txn, 'credit', feedback)
-
-        if dynamic['id'] == 'inexistent' or dynamic['limit'] == 0:
-            score = 0
-
-        else:
-            id = dynamic['id']
-            limit = dynamic['limit']
-
-            # Keep ony transactions in best credit account
-            transat = [x for x in txn if x['account_id'] == id]
-
-            if transat:
-                dates = list()
-                amounts = list()
-                for t in transat:
-                    dates.append(t['date'])
-                    amounts.append(t['amount'])
-                df = pd.DataFrame(
-                    data={'amounts': amounts}, index=pd.DatetimeIndex(dates)
-                )
-
-                # Bin by month credit card 'purchases' and 'paybacks'
-                util = df.groupby(pd.Grouper(freq='M'))['amounts'].agg(
-                    [
-                        ('payback', lambda x: x[x < 0].sum()),
-                        ('purchases', lambda x: x[x > 0].sum()),
-                    ]
-                )
-                util['cred_util'] = [x / limit for x in util['purchases']]
-
-                # Exclude current month
-                if util.iloc[-1, ].name.strftime(
-                    '%Y-%m'
-                ) == datetime.today().date().strftime('%Y-%m'):
-                    util = util[:-1]
-
-                avg_util = np.mean(util['cred_util'])
-                m = np.digitize(len(util) * 30, params['duration'], right=True)
-                n = np.digitize(avg_util, params['credit_util_pct'], right=True)
-                score = params['activity_cns_mtx'][m][n]
-
-                feedback['credit']['utilization_ratio'] = round(avg_util, 2)
-
-            else:
-                raise Exception('no credit history')
-
-    except Exception as e:
-        score = 0
-        feedback['credit']['error'] = str(e)
-
-    finally:
-        return score, feedback
-
-
-# @evaluate_function
-def credit_interest(acc, txn, feedback, params):
+def plaid_credit_metrics(feedback, params, metadata):
```

Comment:
Same here: docstrings needed for a function like this. Especially, considering that it's an open source projects that other developers will possibly fork and edit.